sale_project_user_profile/models/product_template.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    track_service = fields.Selection(
```

Comment:
IMHO I think you should use a new boolean field to maintain compatibility.
If you want I can make a PR to your branch

If this module is installed on any client you can add a script:
`UPDATE new_boolean_field = true, track_service = 'timesheet' WHERE track_service='user_profile'`